### PR TITLE
Clean tenant principals via umb

### PR DESCRIPTION
**Pull request by astrozzc**
## Link(s) to Jira
- https://issues.redhat.com/browse/RHCLOUD-31280

## Description of Intent of Change(s)
Use asynchronies way to clean the deactivated principals 

## Local Testing
Unit tests

## Checklist
- [ ] if API spec changes are required, is the spec updated?
- [ ] are there any pre/post merge actions required? if so, document here.
- [ ] are theses changes covered by unit tests?
- [ ] if warranted, are documentation changes accounted for?
- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?
- [ ] is there known, direct impact to dependent teams/components?
  - [ ] if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: deploy/rbac-clowdapp.yml [hunk 1]
```diff
@@ -3447,6 +3479,18 @@ parameters:
...
   value: 'False'
+- name: PRINCIPAL_CLEANUP_DELETION_ENABLED_UMB
+  description: Allow cleanup job to delete principals via messages from UMB
+  value: 'False'
+- name: UMB_HOST
+  description: Host of the UMB service
+  value: 'localhost'
+- name: UMB_PORT
+  description: Port of the UMB service
+  value: 61612
```
Comment (petracihalova) [inline deploy/rbac-clowdapp.yml:3490]: the value should be string?
```suggestion
  value: '61612'
```

**Review by petracihalova — APPROVED** [review]

File: deploy/rbac-clowdapp.yml (hunk 1 above)
Comment (astrozzc) [reply deploy/rbac-clowdapp.yml]: Nice catch, thanks!

File: deploy/rbac-clowdapp.yml [hunk 2]
```diff
@@ -78,13 +78,29 @@ objects:
...
           name: model-access-permissions
+        - name: umb-cert
+          secret:
+            secretName: service-accounts
+            items:
+            - key: umb-cert
+              path: cert.pem
+        - name: umb-key
+          secret:
+            secretName: service-accounts
```
Comment (petracihalova) [inline deploy/rbac-clowdapp.yml:100]: I believe the value of `secretName` should be unique

Comment (astrozzc) [reply deploy/rbac-clowdapp.yml]: I think you are right, moved them into one

**Pull request merged**